File: typescript/vulnerable3.ts
/**
 * @fileoverview Additional vulnerable TypeScript code samples for SAST testing
 * @author Test Suite
 * @created 2025-01-21
 */

// Hardcoded API keys
const API_KEYS = {
  stripe: 'sk_live_51H3ll0W0rld123456789',
  aws: 'AKIAIOSFODNN7EXAMPLE',
  github: 'ghp_1234567890abcdefghijklmnopqrstuvwxyz'
};

// Path Traversal
import * as fs from 'fs';
function readUserFile(username: string, filename: string): string {
  return fs.readFileSync(`/home/${username}/files/${filename}`, 'utf8');
}

// Weak crypto
import * as crypto from 'crypto';
function hashData(data: string): string {
  return crypto.createHash('md5').update(data).digest('hex');
}

File: go/vulnerable3.go
/**
 * Additional vulnerable Go code samples for SAST testing
 * Author: Test Suite
 * Created: 2025-01-21
 */

package main

import (
	"crypto/md5"
	"fmt"
	"io/ioutil"
	"path/filepath"
)

// Hardcoded API keys
const (
	StripeKey = "sk_live_51H3ll0W0rld123456789"
	AWSKey    = "AKIAIOSFODNN7EXAMPLE"
)

// Weak crypto - MD5
func hashData(data string) string {
	hash := md5.Sum([]byte(data))
	return fmt.Sprintf("%x", hash)
}

// Path Traversal
func readConfigFile(filename string) ([]byte, error) {
	path := filepath.Join("/etc/config", filename)
	return ioutil.ReadFile(path)
}

